imports/api/HandScoreCalculator.js

```diff
@@ -0,0 +1,151 @@
+        selfdraw_delta: jpn_selfdraw_delta,
+        mistake_delta: jpn_mistake_delta
+    }
+};
+
+/**
+ * Calculates the change in points as a result of a deal-in hand
+ * Uses the formula from:
+ * https://en.wikipedia.org/wiki/Japanese_Mahjong_scoring_rules
+ *
+ * @param {Number} points - the total number of points in hand
+ * @param {Number} fu - the fu value of the hand
+ * @param {Number} bonus - the current round bonus
+ * @param {string} dealerWind - the true wind of the current dealer
+ * @param {string} winnerWind - the winning seat true wind
+ * @param {string} loserWind - the losing seat true wind
+ * @param {Number} riichiSticks - the number of riichi sticks won
+ * @return {Object} containing the point difference for each seat
+ */
+function jpn_dealin_delta(points,
+                          fu,
+                          bonus,
+                          dealerWind,
+                          winnerWind,
+                          loserWind,
+                          riichiSticks) {
+    let winds = {};
+    Constants.WINDS.forEach(w => {winds[w] = 0});
+
+    let handValue;
+    let bonusPoints = bonus * Constants.JPN_BONUS_POINTS;
+
+    // The multiplier is for whether or not it's a dealer victory
+    let multiplier = (winnerWind != dealerWind) ? 4 : 6;
+
+    // Check to see if you have to count basic points
+    if (points < 5) {
+        if (fu === 20 || (points === 1 && fu === 25)) {
+            throw TypeError("Invalid points/fu combination");
+        } else {
+            // Calculate hand value, if it's above a mangan, cap it there
+            let manganPayout = Constants.JPN_MANGAN_BASE_POINTS * multiplier;
+            handValue = Math.ceil((fu * Math.pow(2, 2 + points)) * multiplier / 100) * 100;
+            handValue = (handValue > manganPayout) ? manganPayout : handValue;
+        }
+    } else {
+        handValue = manganValue(points) * multiplier;
+    }
+
+    // Add everything together to finalize total
+    winds[winnerWind] = handValue + bonusPoints + (riichiSticks * Constants.JPN_RIICHI_POINTS);
+    winds[loserWind] = -(handValue + bonusPoints);
+
+    return winds;
+};
+
+/**
+ * Calculates the change in points as a result of a self-drawn hand
+ * Uses the formula from https://en.wikipedia.org/wiki/Japanese_Mahjong_scoring_rules
+ *
+ * @param {Number} points - the total number of han + dora
+ * @param {Number} fu - the fu value of the hand
+ * @param {Number} bonus - the current round bonus
+ * @param {string} dealerWind - the true wind of the current dealer
+ * @param {string} winnerWind - the winning seat true wind
+ * @param {Number} riichiSticks - the total number of thrown riichi sticks
+ * @return {Object} containing the point difference for each seat
+ */
+function jpn_selfdraw_delta(points, fu, bonus, dealerWind, winnerWind, riichiSticks) {
+    let winds = {};
+    Constants.WINDS.forEach(w => {winds[w] = 0});
+    let basicPoints;
+    let nonDealerPays;
+    let dealerPays;
```

Comment:
You can declare these uninitialized variables on one line

Reply:
Yo I copy & pasted your code man, will change.